draft-thomson-quic-tls.md

```diff
@@ -307,10 +308,24 @@ The following transitions are defined:
 * The server transitions to using 1-RTT keys after sending its Finished message.
   This causes the KEY_PHASE bit to be set to 0 if early data was accepted, and 1
   if the server rejected early data.
+  TODO: I think you're saying that every key transition result in a change
+  to the value of this bit. That makes sense to me. But I'm not understanding
+  the point about rejecting 0-RTT data. Is that the same thing as rejecting the
+  handshake? I'm guessing not. I think I'm not quite able to follow the
+  different values for this bit in all situations. Perhaps it would make sense
+  to show some example with 0-RTT accepted and rejected.
```

Comment:
There is always the possibility that 0-RTT data has to be rejected by the server.  If the timing is off; if the server has reset and lost its keys; if it decides that it wants to avoid replay; any number of reasons.  That leaves us with the following scenarios:

1. The client doesn't do 0-RTT.  In that case, the ServerHello will be unprotected and KEY_PHASE will be 0.
2. The client does 0-RTT, but the server rejects it.  Like above, the ServerHello will be unprotected and KEY_PHASE will be 0.
3. The client does 0-RTT and the server accepts it.  In this case, the server will use the 0-RTT keys to protect its ServerHello and the KEY_PHASE will be 1 to signal that change.

There is a risk here, and that is that when the server transitions to full encryption after sending all its handshake messages, only those messages make it to the client.  That is, all the server's handshake messages are lost.  In that case, KEY_PHASE on those packets will be 0 again and the client will attempt to decode them as though they were unprotected.  That's a pretty easy thing to detect, so the risk there is low.